pysat/instruments/template_instrument.py

```diff
@@ -0,0 +1,339 @@
+        instantiating an Instrument object. These additional keywords
+        are passed through to this routine by pysat.
+    
+    Returns
+    -------
+    data, metadata
+        Data and Metadata are formatted for pysat. Data is an xarray 
+        DataSet while metadata is a pysat.Meta instance.
+        
+    Note
+    ----
+    Any additional keyword arguments passed to pysat.Instrument
+    upon instantiation are passed along to this routine.
+    
+    Examples
+    --------
+    ::
+        inst = pysat.Instrument('ucar', 'tiegcm')
+        inst.load(2019,1)
+    
+    """
+
+    # netCDF4 files, particularly those produced
+    # by pysat can be loaded using a pysat provided
+    # function
+    # Metadata in our notional example file is
+    # labeled by strings determined by a standard
+    # we can adapt pysat to the standard by specifying
+    # the string labels used in the file
+    # function below returns both data and metadata
+    return pysat.utils.load_netcdf4(fnames, epoch_name='Epoch', 
+                                    units_label='Units', name_label='Long_Name', 
+                                    notes_label='Var_Notes', desc_label='CatDesc',
+                                    plot_label='FieldNam', axis_label='LablAxis', 
+                                    scale_label='ScaleTyp',
+                                    min_label='ValidMin', max_label='ValidMax',
+                                    fill_label='FillVal')
+    
+
+    # This code below demonstrates the use of xarray
+    # functions to load TIEGCM data
+    # Metadata is transferred from xarray to the Instrument object
+    # Data is transferred as well
+    # data not indexed by time are transferred to the Instrument object as an attribute
+        
+    # load data
+    data = xr.open_dataset(fnames[0])
+    # move attributes to the Meta object
+    # these attributes will be trasnferred to the Instrument object
+    # automatically by pysat
+    meta = pysat.Meta()
+    for attr in data.attrs:
+        setattr(meta, attr[0], attr[1])
+    data.attrs = []
+        
+    # fill Meta object with variable information
+    for key in data.variables.keys():
+        attrs = data.variables[key].attrs
+        meta[key] = attrs
+
+    # move misc parameters from xarray to the Instrument object via Meta
+    # doing this after the meta ensures all metadata is still kept
+    # even for moved variables
+    meta.p0 = data['p0']
+    meta.p0_model = data['p0_model']
+    meta.grav = data['grav']
+    meta.mag = data['mag']
+    meta.timestep = data['timestep']
+    # remove these variables from xarray
+    data = data.drop(['p0', 'p0_model', 'grav', 'mag', 'timestep'])
+            
+    return data, meta
+
+
+def list_files(tag=None, sat_id=None, data_path=None, format_str=None):
+    """Produce a list of files corresponding to PLATFORM/NAME.
+
+    This routine is invoked by pysat and is not intended for direct 
```

Comment:
Should these extra comments be included in the bottom in a section called "Notes"?